### ISSUE TITLE
Openai-L-14-336

### ISSUE THREAD

**Issue by LKELN** [microsoft/LLM2CLIP#19]
Expected tensor for argument #1 'indices' to have one of the following scalar types: Long, Int; but got torch.cuda.FloatTensor instead (while checking arguments for embedding) 

input_pixels = processor(images=image, return_tensors="pt").pixel_values.to('cuda')
text_features = l2v.encode(captions, convert_to_tensor=True).to("cuda")
with torch.no_grad(), torch.cuda.amp.autocast():
    image_features = model.get_image_features(input_pixels)
    text_features = model.get_text_features(text_features)

    image_features /= image_features.norm(dim=-1, keepdim=True)
    text_features /= text_features.norm(dim=-1, keepdim=True)

    text_probs = (100.0 * image_features @ text_features.T).softmax(dim=-1)

print("Label probs:", text_probs)

When I try to use [LLM2CLIP-Openai-L-14-336](https://huggingface.co/microsoft/LLM2CLIP-Openai-L-14-336). this erro is appear？Can you fix it？

**Comment by raytrun** [comment]
Could you provide a more complete code? The example given [here](https://huggingface.co/microsoft/LLM2CLIP-Openai-L-14-336) is able to run correctly.

**Comment by LKELN** [comment]
processor = CLIPImageProcessor.from_pretrained("/group/40048/keningliu/tools/models/clip-vit-large-patch14-336")
model_name_or_path = "/group/40048/keningliu/tools/models/LLM2CLIP-Openai-L-14-336" # or /path/to/local/LLM2CLIP-Openai-L-14-336
model = AutoModel.from_pretrained(
    model_name_or_path, 
    torch_dtype=torch.bfloat16,
    trust_remote_code=False).to('cuda').eval()
print(model)
print(type(model).__mro__)
captions = ["a diagram", "a dog", "a cat"]
image_path = "/group/40048/keningliu/tools/FlagData/pipeline.png"

image = Image.open(image_path)
input_pixels = processor(images=image, return_tensors="pt").pixel_values.to('cuda')
text_features = l2v.encode(captions, convert_to_tensor=True).to('cuda')
print(model)
with torch.no_grad(), torch.cuda.amp.autocast():
    image_features = model.get_image_features(input_pixels)
    text_features = model.get_text_features(text_features)

    image_features /= image_features.norm(dim=-1, keepdim=True)
    text_features /= text_features.norm(dim=-1, keepdim=True)

    text_probs = (100.0 * image_features @ text_features.T).softmax(dim=-1)

print("Label probs:", text_probs)

**Comment by LKELN** [comment]
I found that when loading the model, the weight files mentioned in your paper were not loaded. Instead, only the CLIP structure was loaded, which makes it impossible to conduct inference.

**Comment by raytrun** [comment]
The model should be loaded in the following manner:
model = AutoModel.from_pretrained(model_name_or_path,  torch_dtype=torch.float16, trust_remote_code=True).to('cuda').eval()

trust_remote_code=True is necessary.

**Comment by LKELN** [comment]
Thanks
